File: 期末考2.0/Q5 Python Student Inheritance/Student.py
from Person import Person

class Student(Person):
    def __init__(self, newName, newMathScore, newChineseScore, newBirthday):
        self.__mathScore = newMathScore
        self.__chineseScore = newChineseScore
        super().__init__(newName, newBirthday)
    def setMathScore(self, newMathScore):
        self.__mathScore = newMathScore
    def setChineseScore(self, newChineseScore):
        self.__chineseScore = newChineseScore
    def getMathScore(self):
        return self.__mathScore
    def getChineseScore(self):
        return self.__chineseScore
    def average(self):
        return (self.__mathScore + self.__chineseScore) / 2
    def toString(self):
        super().toString()
        print(self.average())

File: 期末考2.0/Q2 Student Polymorphism/Student.cpp
#include"Student.h"

Student::Student(string newName, int newMathScore, int newChineseScore, Date newBirthday) {
    mathScore = newMathScore;
    chineseScore = newChineseScore;
    setName(newName);
    setBirthday(newBirthday);
}
void Student::setMathScore(int newMathScore) {
    mathScore = newMathScore;
}
void Student::setChineseScore(int newChineseScore) {
    chineseScore = newChineseScore;
}
int Student::getMathScore() {
    return mathScore;
}
int Student::getChineseScore() {
    return chineseScore;
}
double Student::average() {
    return (mathScore + chineseScore) / 2.0;
}
string Student::toString(){
    stringstream ss;
    ss << setprecision(1) << fixed;
    ss << getName() << " " << getBirthday().toString() << " " << average() << endl;
    return ss.str();
}
